src/plugins/explore/public/components/query_panel/components/editor_stack/index.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { PromptEditor } from './prompt_editor';
+import { QueryEditor } from './query_editor';
+import { LanguageType } from './shared';
+
+interface EditorStackProps {
+  onPromptChange: (value: string) => void;
+  onQueryChange: (value: string) => void;
+  languageType: LanguageType;
+  isDualEditor: Boolean;
+  isPromptReadOnly: boolean;
+  isEditorReadOnly: boolean;
+  handleQueryEdit: () => void;
+  handlePromptEdit: () => void;
+  handleQueryRun: (query?: string) => void;
```

Comment:
these are a lot of props. wouldn't it be easier to pass around the query object? and i see the query is optional is that correct?

examples
```
handleRun: (query: Query) => void;

// ex. NLQ
handleRun(
   { query: "How many 400s?", language: "NLQ" }
)

// ex. PPL
handleRun(
  { query: "source = *", language: "PPL" }
)
```

then in the logic you could utilize a switch case

Reply:
historically the coding style i have seen in the repo, for event handler naming conventions:

- `handle[Event]` for internal class/component methods that process events
- `on[Event]` for props that are passed down to child components

Example
```typescript
class MyComponent {
  handleSubmit() { ... }
  handleFilterChange() { ... }
  
  render() {
    return (
      <ChildComponent 
        onSubmit={this.handleSubmit}
        onChange={this.handleFilterChange}
      />
    );
  }
}
```

for consistency and alignment it might be better